x-pack/plugins/actions/server/create_execute_function.ts

```diff
@@ -31,6 +31,7 @@ interface CreateExecuteFunctionOptions {
 export interface ExecuteOptions
   extends Pick<ActionExecutorOptions, 'params' | 'source' | 'relatedSavedObjects' | 'consumer'> {
   id: string;
+  uuid?: string;
```

Comment:
Curious why this is optional.  I think it's because for really old rules that haven't been updated since we added the `uuid` to the actions, it won't be there, and we don't have a migration.  But double-checking in case it's BWC or something.

Just from a quick scan of the code, it doesn't seem like we added many tests in this PR that would have a non-existing `uuid`.  Maybe some of the tests already deal with this, that I wouldn't have noticed?

I see one reference to a `uuid!`, which was just refactor/move from old code - but ... scares me :-). As I take a closer look at this PR, maybe I'll figure out it's ok.

Reply:
It is typed as optional even though it should always exist on the action :(. I think because we use a single schema in the API and the persistence layer and we don't expect the UUID to be passed into the API but we create one before it reaches the persistence layer.

Reply:
I found the PR where we added the `uuid`.  For some reason I thought we hadn't done a migration to add to old rules, but we did! [Add uuid to rule actions](https://github.com/elastic/kibana/pull/148038).  My fear was we didn't, so an upgrade from an old version wouldn't have `uuid`s, which got me wondering about the cases where you check it.

Seems like something we should fix.  Open an issue?  No need to fix in this PR (and may be nasty, presumably another type to deal with!).

Reply:
Created an issue: https://github.com/elastic/kibana/issues/195255